common/src/main/java/com/microsoft/identity/common/internal/cache/MicrosoftFamilyOAuth2TokenCache.java

```diff
@@ -91,107 +85,75 @@ public ICacheRecord loadByFamilyId(@Nullable final String clientId,
-        }
-
-        // If there is no RT for this app, try to find any RT in the family (family id ONLY)
-        if (null == result || null == result.getRefreshToken()) {
-            Logger.warn(
-                    TAG + methodName,
-                    "Matching RT could not be found. Searching for compatible FRT."
-            );
-
-            final List<Credential> allCredentials = getAccountCredentialCache().getCredentials();
-
-            // The following fields must match:
-            // - environment
-            // - home_account_id
-            // - credential_type == RT
-            //
-            // The following fields do not matter:
-            // - clientId doesn't matter (FRT)
-            // - target doesn't matter (FRT)
-            // - realm doesn't matter (MRRT)
-
-            final List<RefreshTokenRecord> allRefreshTokens = new ArrayList<>();
-
-            // First, filter down to only the refresh tokens...
-            for (final Credential credential : allCredentials) {
-                if (credential instanceof RefreshTokenRecord) {
-                    allRefreshTokens.add((RefreshTokenRecord) credential);
-                }
+        // Try to find any RT in the family...
+        final List<Credential> allCredentials = getAccountCredentialCache().getCredentials();
+
+        // The following fields must match:
+        // - environment
+        // - home_account_id
+        // - credential_type == RT
+        //
+        // The following fields do not matter:
+        // - clientId doesn't matter (FRT)
+        // - target doesn't matter (FRT)
+        // - realm doesn't matter (MRRT)
+
+        final List<RefreshTokenRecord> allRefreshTokens = new ArrayList<>();
+
+        // First, filter down to only the refresh tokens...
+        for (final Credential credential : allCredentials) {
+            if (credential instanceof RefreshTokenRecord) {
+                allRefreshTokens.add((RefreshTokenRecord) credential);
             }
+        }
 
-            Logger.info(
-                    TAG + methodName,
-                    "Found [" + allRefreshTokens.size() + "] RTs"
-            );
+        Logger.info(
+                TAG + methodName,
+                "Found [" + allRefreshTokens.size() + "] RTs"
+        );
 
-            // Iterate over those refresh tokens and see if any are in the family...
-            final List<RefreshTokenRecord> familyRefreshTokens = new ArrayList<>();
+        // Iterate over those refresh tokens and see if any are in the family...
+        final List<RefreshTokenRecord> familyRefreshTokens = new ArrayList<>();
 
-            for (final RefreshTokenRecord refreshToken : allRefreshTokens) {
-                if (refreshToken.getFamilyId().equals(familyId)) {
-                    familyRefreshTokens.add(refreshToken);
-                }
+        for (final RefreshTokenRecord refreshToken : allRefreshTokens) {
+            if (refreshToken.getFamilyId().equals(familyId)) {
```

Comment:
Can we check for familyId and other properties in the for loop above?

Reply:
I'll collapse these, thanks

Reply:
Resolved in:
- 47954879d4df738d726321a9a8110a29669f0954
- abec7f902e22f1e2ad4772c80f4d2d992d1ba2b7